skpro/regression/cyclic_boosting.py

```diff
@@ -0,0 +1,315 @@
+        self.interaction = interaction
+        self.quantiles = [0.2, 0.5, 0.8]
+        self.quantile_values = []
+        self.quantile_est = []
+        self.qpd = None
+        self.distr_type = distr_type
+
+        super().__init__()
+
+        # check parameters
+        if not isinstance(feature_properties, dict):
+            raise ValueError("feature_properties must be dict")
+        for i in interaction:
+            if not isinstance(i, tuple):
+                raise ValueError("interaction must be tuple")
+
+        # build estimators
+        features = list(self.feature_properties.keys())
+        for i in interaction:
+            features.append(i)
+
+        for quantile in self.quantiles:
+            self.quantile_est.append(
+                pipeline_CBMultiplicativeQuantileRegressor(
+                    quantile=quantile,
+                    feature_properties=self.feature_properties,
+                    feature_groups=features,
+                    maximal_iterations=50,
+                )
+            )
+
+    def _fit(self, X, y):
+        """Fit regressor to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : pandas DataFrame
+            feature instances to fit regressor to
+        y : pandas DataFrame, must be same length as X
+            labels to fit regressor to
+
+        Returns
+        -------
+        self : reference to self
+        """
+
+        self._y_cols = y.columns
+        y = y.to_numpy().flatten()
+
+        # multiple quantile regression for full probability estimation
+        for est in self.quantile_est:
+            est.fit(X.copy(), y)
+
+        return self
+
+    def _predict(self, X):
+        """Predict labels for data from features.
+
+        State required:
+            Requires state to be "fitted" = self.is_fitted=True
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+
+        Parameters
+        ----------
+        X : pandas DataFrame, must have same columns as X in `fit`
+            data to predict labels for
+
+        Returns
+        -------
+        y : pandas DataFrame, same length as `X`, same columns as `y` in `fit`
+            labels predicted for `X`
+        """
+
+        index = X.index
+        y_cols = self._y_cols
+
+        # quantile prediction
+        for est in self.quantile_est:
+            yhat = est.predict(X.copy())
+            self.quantile_values.append(yhat)
+
+        j_qpd = J_QPD_S(
+            0.2,
+            qv_low=self.quantile_values[0],
+            qv_median=self.quantile_values[1],
+            qv_high=self.quantile_values[2],
+            index=index,
+            columns=y_cols,
+        )
+        self.qpd = j_qpd
+
+        y_pred = pd.DataFrame(j_qpd.mean(), index=index, columns=y_cols)
+        return y_pred
+
+    def _predict_proba(self, X):
```

Comment:
I would suggest to also use predict_quantiles for the quantile predictions. And here we can then just give the QPD.

Reply:
hm, that's an interesting question. `predict_quantiles` will automatically default to the `quantiles` method of the `predict_proba` return, that ensures consistency. The interface does not enforce exact consistency, so if there's two consistent options, a method/strategy parameter in the `__init__` may be the way to go.

Reply:
Ok, I see. In that case, let's stay consistent. Let me summarize my suggestion in the other comment thread :).